examples/networklayer_tcp.c

```diff
@@ -450,6 +450,8 @@ UA_ServerNetworkLayer ServerNetworkLayerTCP_new(UA_ConnectionConfig conf, UA_UIn
 
 static UA_StatusCode ClientNetworkLayerGetBuffer(UA_Connection *connection, UA_ByteString *buf) {
 #ifndef UA_MULTITHREADING
+    if(connection->state == UA_CONNECTION_CLOSED)
+        return UA_STATUSCODE_BADCONNECTIONCLOSED;
```

Comment:
in all our own code there is 
`UA_ByteString message;`
before the call of `getbuffer(&c->connection, &message)`. 
IMO this is the responsibility of the caller.